webidl.txt

```diff
@@ -72,50 +46,104 @@ dictionary AudioDecoderParams {
   MediaTime? codecDelay;
 } 
 
-[Constructor()]
-interface AudioTrackWriter {
-  readonly attribute WritableStream writable; // of DecodedAudioPacket
-  readonly attribute MediaStreamTrack track;
+dictionary AudioDecoderInput {
+  Uint8Array data;
+  MediaTime timestamp;
 }
 
+dictionary AudioDecoderOutput {
+  AudioBuffer buffer;
+  // TODO: decode stats.
+}
 
-[Constructor(MediaStreamTrack track)]
-interface VideoTrackReader {
-  readonly attribute ReadableStream readable; // of DecodedVideoFrame
+dictionary AudioEncoderStaticSettings {
+  // probably opus-specific
+  bool fec = false;  // enabled or not
+  bool dtx = false;  // enabled or not
+  bool cbr = false;  // cbr or not (vbr if not)
+  bool speechMode = false;  // speech-specific mode or not
+}
+
+dictionary AudioEncoderDynamicSettings {
+  // not supported by all codecs
+  // null/unset means use the codec default
+  unsigned long? bitsPerSecond;  
+
+  // codec-specific
+  // null/unset means use the codec default
+  unsigned long? complexity;
 }
 
-interface DecodedVideoFrame {
+dictionary AudioEncoderSettings : AudioEncoderStaticSettings, AudioEncoderDynamicSettings {
+}
```

Comment:
Would having the AudioEncoderSettings contain the other two rather than inherit from them be better?

Reply:
I wanted to keep the JSON from becoming too hierarchical. I don't think it's too difficult to keep the names from clashing.